File: app/src/main/java/com/example/amit/uniconnexample/SnackbarBehavior.java
package com.example.amit.uniconnexample;

import android.content.Context;
import android.support.design.widget.CoordinatorLayout;
import android.support.design.widget.Snackbar;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by amit on 9/12/16.
 */

public class SnackbarBehavior extends CoordinatorLayout.Behavior<ViewGroup>{

    public SnackbarBehavior(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean layoutDependsOn(CoordinatorLayout parent, ViewGroup child, View dependency) {
        return dependency instanceof Snackbar.SnackbarLayout;
    }

    @Override
    public boolean onDependentViewChanged(CoordinatorLayout parent, ViewGroup child, View dependency) {
        CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) child.getLayoutParams();
        params.bottomMargin = parent.getHeight()- (int) dependency.getY();
        child.setLayoutParams(params);
        return true;
    }

    @Override
    public void onDependentViewRemoved(CoordinatorLayout parent, ViewGroup child, View dependency) {
        CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) child.getLayoutParams();
        params.bottomMargin = 0;
        child.setLayoutParams(params);
    }
}

File: app/src/main/java/com/example/amit/uniconnexample/SnackbarBehavior.kt
package com.example.amit.uniconnexample

import android.content.Context
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.google.android.material.snackbar.Snackbar
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup

/**
 * Created by amit on 29/04/18.
 */

class SnackbarBehavior(context: Context, attrs: AttributeSet) : CoordinatorLayout.Behavior<ViewGroup>(context, attrs) {

    override fun layoutDependsOn(parent: CoordinatorLayout, child: ViewGroup, dependency: View): Boolean {
        return dependency is Snackbar.SnackbarLayout
    }

    override fun onDependentViewChanged(parent: CoordinatorLayout, child: ViewGroup, dependency: View): Boolean {
        val params = child.layoutParams as CoordinatorLayout.LayoutParams
        params.bottomMargin = parent.height - dependency.y.toInt()
        child.layoutParams = params
        return true
    }

    override fun onDependentViewRemoved(parent: CoordinatorLayout, child: ViewGroup, dependency: View) {
        val params = child.layoutParams as CoordinatorLayout.LayoutParams
        params.bottomMargin = 0
        child.layoutParams = params
    }
}